src/linter/conf.go

```diff
@@ -68,6 +75,82 @@ type Config struct {
 	StrictMixed bool
 }
 
+type RuleNode struct {
+	Children map[string]*RuleNode // Child nodes (subdirectories and files)
+	Enable   map[string]bool      // Rules enabled at this level
+	Disable  map[string]bool      // Disabled rules at this level
+}
+
+func NewRuleNode() *RuleNode {
+	return &RuleNode{
+		Children: make(map[string]*RuleNode),
+		Enable:   make(map[string]bool),
+		Disable:  make(map[string]bool),
+	}
+}
+
+type PathRuleSet struct {
+	Enable  map[string]bool // Rules that are enabled for this path
+	Disable map[string]bool // Rules that are disabled for this path
+}
+
+func BuildRuleTree(pathRules map[string]*PathRuleSet) *RuleNode {
+	root := NewRuleNode()
+
+	for path, ruleSet := range pathRules {
+		normalizedPath := filepath.ToSlash(filepath.Clean(path))
+		parts := strings.Split(normalizedPath, "/")
+		currentNode := root
+
+		for _, part := range parts {
+			if part == "" {
+				continue
+			}
+			if _, exists := currentNode.Children[part]; !exists {
+				currentNode.Children[part] = NewRuleNode()
+			}
+			currentNode = currentNode.Children[part]
+		}
+
+		for rule := range ruleSet.Enable {
+			currentNode.Enable[rule] = true
+		}
+		for rule := range ruleSet.Disable {
+			currentNode.Disable[rule] = true
+		}
+	}
+
+	return root
+}
+
+func IsRuleEnabled(root *RuleNode, filePath string, checkRule string) bool {
+	normalizedPath := filepath.ToSlash(filepath.Clean(filePath))
```

Comment:
1. let's move it to `root.go`
2. Can we definitely have a situation where we need to apply `filepath.Clean`?

Reply:
1. Done
2. I think that this is a good practice to combine it:
2.1. filePath := "A//B///C/file.php" will give us parts = ["A", "", "B", "", "", "C", "file.php"] => this is not very good
2.2. filePath := "A/B/C/" => parts = ["A", "B", "C", ""]
Or some another examples. I think that is ok to use it

Reply:
1. I don't see it( `filePath = filepath.ToSlash(filepath.Clean(relativePath))`
2. ok

Reply:
1. Why? Its okey that we can have 'bad' path or something else from the world. Function will guarantee that path would be normalize, because someone could forget about it. Why? It’s okay that we might receive a ‘bad’ path or something else from the external world. The function will guarantee that the path is normalized, because someone could forget to do it.